liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/BaseDevTest.java

```diff
@@ -0,0 +1,215 @@
+/*******************************************************************************
+ * (c) Copyright IBM Corporation 2019.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package net.wasdev.wlp.test.dev.it;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.maven.shared.utils.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class BaseDevTest {
+
+   File tempProj;
+   File basicDevProj;
+   File logFile;
+   boolean isWindows = false;
+
+   @Test
+   public void testing() throws Exception {
+      assertTrue(true);
+   }
+  
+   @Before
+   public void setUp() throws Exception {
+      String os = System.getProperty("os.name");
+      if (os != null && os.toLowerCase().startsWith("windows")) {
+         isWindows = true;
+      }
+      
+      if (!isWindows) { // skip tests on windows until server.env bug is fixed
+         
+         tempProj = Files.createTempDirectory("temp").toFile();
+         assertTrue(tempProj.exists());
+         
+         basicDevProj = new File("../resources/basic-dev-project");
+         assertTrue(basicDevProj.exists());
+
+         FileUtils.copyDirectoryStructure(basicDevProj, tempProj);
+         assertTrue(tempProj.listFiles().length > 0);
+         
+         logFile = new File(basicDevProj, "/logFile.txt");
+         logFile.createNewFile();
+         
+         String pluginVersion = System.getProperty("testing");
```

Comment:
move this logic for replacing the plugin version into a method for better readability